### ISSUE TITLE
RANDOM.C and use of asm ("RDTSC" : "=A" (result)) FAILS on ARM

### ISSUE THREAD

**Issue by LittleDuke** [Minege/eSSP#12]
in "lib/random.c" on line 146

asm ("RDTSC" : "=A" (result));

Does not compile on ARM/ARM64 based Raspberry PI computers for obvious reasons.

A better more cross-platform compatible approach would be to support something like "rng-tools"

https://wiki.archlinux.org/index.php/Rng-tools

**Comment by Minege** [comment]
Hi @LittleDuke,
Can you try to again on your Raspberry Pi, but install the library like this instead:
`pip install git+https://github.com/Minege/eSSP.git@update` ?

**Comment by LittleDuke** [comment]
I solved the problem by reading from /dev/urandom

Can upload patch

-dvd

**Comment by ymaktepi** [comment]
Hi @LittleDuke 

I'd be interested in seeing your patch as I'm currently running into the same issue. Would you be kind enough to create a PR if you still have the updated code around? Thanks in advance

@Minege 
Long time no see, how's life? I tried compiling the C part of the lib on a raspberry pi 4 (edit: using the `update` branch), and it seems to still crash at the same point:
```
Random.c: In function ‘GetRTSC’:
Random.c:141:2: error: impossible constraint in ‘asm’
  asm ("RDTSC" : "=A" (result));
  ^~~
make: *** [Makefile:32: Random.o] Error 1
```
I did not try on a RPi3 but I could if it's needed.

**Comment by ymaktepi** [comment]
Should be fixed in #15
